Youngdo212

youngdo212 opened the pull request.

Commit:
수정(1차). 다각형의 넓이 함수

Commit:
수정(1차). 다각형의 넓이 함수

Commit:
수정(2차). 다각형의 넓이 함수. getArea함수 추가

Commit:
수정(2차). 다각형의 넓이 함수. getArea함수 추가

Commit:
수정(3차). 다각형의 넓이 함수. 로깅을 위한 함수(getReport), 전역변수 추가

Commit:
수정(3차). 다각형의 넓이 함수. 로깅을 위한 함수(getReport), 전역변수 추가

Commit:
Update markdown document

한 눈에 알아볼 수 있도록 md문서를 수정했습니다.

Commit:
Update markdown document

한 눈에 알아볼 수 있도록 md문서를 수정했습니다.

Commit:
Refactor getArea function using spread operator

Commit:
Refactor getArea function using spread operator

Commit:
Refactor gerArea function using closure

closure를 사용해 전역변수를 삭제하였습니다.
markdown파일을 변경된 내용에 맞게 수정하였습니다.

crongro commented on 180307_polygonArea.js:
```diff
@@ -1,29 +1,87 @@
+    switch(arguments.length){
+        case 1:
+            if(typeof arguments[0] !== "number"){
+                console.log("숫자형 타입만 계산이 가능합니다");
```
여기서 오류내용을 반환하거나, 오류를 발생시켜서 프로그램을 멈추게 하는 게 나을 거 같아요.

즉,
return을 하거나, 
throw를 하거나.

Commit:
Change code using throw and return

throw와 return을 사용해 코드를  수정

Commit:
Merge commit

crongro commented on 180307_polygonArea.js:
```diff
@@ -1,19 +1,47 @@
     switch(arguments.length){
         case 1:
             if(typeof arguments[0] !== "number"){
+<<<<<<< HEAD
```
충돌을 해결하지 않고 보내셨???

crongro reviewed (changes requested):
git 충돌시 어떻게 소스코드를 수정해야 하는지 알아보세요~

youngdo212 committed:
Resolve conflict

crongro commented on 180307_polygonArea.js:
```diff
@@ -1,29 +1,94 @@
-        console.log("하나의 인자가 필요합니다");
-        return;
+function circleArea(){
+    switch(arguments.length){
```
실제로는 아래처럼 파라미터를 지정하는게 일반적이겠죠. 
function circleArea(radius){ }

파라미터 없이 코딩하는건 좋지 않을 겁니다. 참고하세요.

crongro commented on 180307_polygonArea.js:
```diff
@@ -1,29 +1,94 @@
+    }
+}
+
+let getArea = (function(){
```
이렇게 즉시함수를 실행한 이유가 뭐에요? 큰 장점이 안보이는데 궁금하네요.

youngdo212 replied:
클로저를 사용하기 위해 제작한 함수를 즉시실행 하였는데요. 이 함수 사용이 일회성에 그칠것 같아 따로 선언은 안해줬습니다. 이름 짓는게 어렵기도 하구요.

crongro reviewed (changes requested):
타입과 인자체크가 꼼꼼하게 했네요.
실제로 모든 함수에서 이렇게하진 않겠지만요.

질문은 답변해주시고, 
리뷰참고하세요.

다음스텝으로 가보세요.

youngdo212 committed:
Refactor functions

기능을 최소화 한 함수들로 만듬
에러체크를 checkError함수로 통일시킴
원의 넓이 구하는 함수를 인자 1개와 인자 2개일 때로 함수 분할
분할 된 원 넓이 함수를 getCircleArea에 넣어줌

youngdo212 reviewed (commented):
함수를 기능별로 분할하였습니다

crongro commented on 180307_polygonArea.js:
```diff
@@ -1,29 +1,71 @@
-        return;
-    }
+function checkError(arr, ...validLength){
+    if(!validLength.includes(arr.length)) throw "올바른 인자 개수를 입력하십시오";
```
includes 함수의 브라우저 호환성을 알아보세요.
지원하지 않는 브라우저에서는 어떻게 처리할 수 있을까요?

youngdo212 replied:
인터넷 익스플로러(데스크탑버전)에서 include 메소드를 지원하지 않습니다(모바일은 가능).
모든 브라우저에서 지원하는 indexOf메소드를 이용할 수 있습니다.

crongro replied:
includes 코드를 그대로 사용하면서, 하위호환성을 보장하는 방법은 무엇일까요? 
polyfill 이란 무엇인지 찾아보고 다시 적용해보세요.

crongro commented on 180307_polygonArea.js:
```diff
@@ -1,29 +1,71 @@
-            return;
+function getManyCircleArea(start, end){
+    let result = [];
+    for(let radius = start; radius <= end; radius++){
```
reduce 를 써서 for문을 대신해볼래요?

crongro commented on 180307_polygonArea.js:
```diff
@@ -1,29 +1,71 @@
+}
+
+function getCircleArea(){
+    checkError(Array.from(arguments), 1, 2);
```
arguments 자체를 전달하기보다는, 배열로 변경하고 그 값을 활용.

crongro reviewed (changes requested):
수정해보세요.

youngdo212 committed:
Refactor some functions

checkError함수의 include대신 indexOf사용(브라우저 호환성)
getManyCircleArea함수를 reduce메소드를 사용해서 변경
checkError함수에 배열을 인자로 넘겨줄 떄 별도로 변수를 정의해 넘김

crongro commented on 180307_polygonArea.js:
```diff
@@ -1,29 +1,74 @@
+    for(let i = start; i <= end; i++){
+        result.push(i);
     }
+    return result.reduce((previous, current) => [...previous, getACircleArea(current)], []);
```
reduce를 직접 한번 구현해보세요.
그리고 현재 코드를 본인이 구현한 코드로 대치해보세요. (화이팅)

crongro reviewed (changes requested):
약간 어려운 미션포함해서~

youngdo212 committed:
Create reduce function myself

reduce()함수를 제작해서 사용

youngdo212 committed:
Create includes method for compatibillity

includes()메소드의 호환성을 위해 새로 정의

crongro reviewed (changes requested):
이런, 
모르는 코드를 구현하지 마세요.

includes 는 직접 아는 범위내에서 구현해서 추가해주세요.

youngdo212 committed:
Recreate includes function

자체 제작한 includes 함수를 사용하였습니다

crongro reviewed (approved):
네. 이렇게 커스트마이징하게 만든 함수들은 
학습용으로 저장해두었다가 미션을 할때마다 재사용해서 써보세요.

문제를 발견하게 되고, 필요하면 기능을 추가하고 그러면서 함수활용도가 더 높아질 수 있을거에요.
